### PR TITLE
570 spdb memtable use seek parralel threshold mishandled

### PR DESCRIPTION
**Pull request by ayulas**

### CODE REVIEW

File: plugin/speedb/memtable/hash_spd_rep.cc [hunk 1]
```diff
@@ -543,32 +543,35 @@ MemTableRep::Iterator* HashSpdRep::GetIterator(Arena* arena) {
...
-     {0, OptionType::kSizeT, OptionVerificationType::kNormal,
-      OptionTypeFlags::kDontSerialize /*Since it is part of the ID*/}},
-    {"use_seek_parralel_threshold",
-     {0, OptionType::kBoolean, OptionVerificationType::kNormal,
-      OptionTypeFlags::kDontSerialize /*Since it is part of the ID*/}},
+static std::unordered_map<std::string, OptionTypeInfo> hash_spdb_factory_info =
+    {
+
+        {"hash_bucket_count",
+         {0, OptionType::kSizeT, OptionVerificationType::kNormal,
```
Comment (Yuval-Ariel) [inline plugin/speedb/memtable/hash_spd_rep.cc:550]: offset should be offsetof(struct HashSpdbRepOptions, hash_bucket_count)

File: plugin/speedb/memtable/hash_spd_rep.cc [hunk 2]
```diff
@@ -543,32 +543,35 @@ MemTableRep::Iterator* HashSpdRep::GetIterator(Arena* arena) {
...
+          OptionTypeFlags::kDontSerialize /*Since it is part of the ID*/}},
+        {"use_seek_parralel_threshold",
+         {0, OptionType::kBoolean, OptionVerificationType::kNormal,
```
Comment (Yuval-Ariel) [inline plugin/speedb/memtable/hash_spd_rep.cc:553]: same here, 
offset should be offsetof(struct HashSpdbRepOptions, use_seek_parralel_threshold)

File: plugin/speedb/memtable/hash_spd_rep.cc [hunk 3]
```diff
@@ -543,32 +543,35 @@ MemTableRep::Iterator* HashSpdRep::GetIterator(Arena* arena) {
...
-
-    {"hash_bucket_count",
-     {0, OptionType::kSizeT, OptionVerificationType::kNormal,
-      OptionTypeFlags::kDontSerialize /*Since it is part of the ID*/}},
-    {"use_seek_parralel_threshold",
-     {0, OptionType::kBoolean, OptionVerificationType::kNormal,
-      OptionTypeFlags::kDontSerialize /*Since it is part of the ID*/}},
+static std::unordered_map<std::string, OptionTypeInfo> hash_spdb_factory_info =
+    {
+
```
Comment (Yuval-Ariel) [inline plugin/speedb/memtable/hash_spd_rep.cc:548]: remove space

File: plugin/speedb/memtable/hash_spd_rep.cc (hunk 2 above)
Comment (ayulas) [reply plugin/speedb/memtable/hash_spd_rep.cc]: done

File: plugin/speedb/memtable/hash_spd_rep.cc (hunk 1 above)
Comment (ayulas) [reply plugin/speedb/memtable/hash_spd_rep.cc]: done

**Review by Yuval-Ariel — APPROVED** [review]

**Comment by ayulas** [comment]
done :-)

File: plugin/speedb/memtable/hash_spd_rep.cc [hunk 4]
```diff
@@ -543,32 +543,36 @@ MemTableRep::Iterator* HashSpdRep::GetIterator(Arena* arena) {
...
-    {"use_seek_parralel_threshold",
-     {0, OptionType::kBoolean, OptionVerificationType::kNormal,
-      OptionTypeFlags::kDontSerialize /*Since it is part of the ID*/}},
+static std::unordered_map<std::string, OptionTypeInfo> hash_spdb_factory_info =
+    {
+        {"hash_bucket_count",
+         {offsetof(struct HashSpdbRepOptions, hash_bucket_count),
+          OptionType::kSizeT, OptionVerificationType::kNormal,
+          OptionTypeFlags::kNone}},
+        {"use_seek_parralel_threshold",
```
Comment (mrambacher) [inline plugin/speedb/memtable/hash_spd_rep.cc:552]: parallel is the proper English spelling

**Review by mrambacher — CHANGES_REQUESTED** [review]
"Parallel", not "parralel" please "-)

**Pull request merged**